patches/tModLoader/Terraria/ModLoader/ModContent.cs

```diff
@@ -491,7 +492,7 @@ public static class ModContent
 			Config.ConfigManager.Unload();
 			CustomCurrencyManager.Initialize();
 			EffectsTracker.RemoveModEffects();
-			Main.MapIcons = new MapIconOverlay().AddLayer(new SpawnMapLayer()).AddLayer(new TeleportPylonsMapLayer()).AddLayer(Main.Pings);
+			Main.MapIcons = new MapIconOverlay().AddLayer(new SpawnMapLayer()).AddLayer(new TeleportPylonsMapLayer()).AddLayer(new ModdedPylonsMapLayer()).AddLayer(Main.Pings);
```

Comment:
See other comment about the map layer, I'd prefer not to have to keep this in sync with `Main.MapIcons`